tests/unit/sagemaker/jumpstart/model/test_model.py

```diff
@@ -441,6 +442,63 @@ def test_eula_gated_conditional_s3_prefix_metadata_model(
             ],
         )
 
+    @mock.patch("sagemaker.utils.sagemaker_timestamp")
+    @mock.patch("sagemaker.jumpstart.utils.validate_model_id_and_get_type")
+    @mock.patch("sagemaker.jumpstart.factory.model.Session")
+    @mock.patch("sagemaker.jumpstart.accessors.JumpStartModelsAccessor.get_model_specs")
+    @mock.patch("sagemaker.jumpstart.model.Model.__init__")
+    @mock.patch("sagemaker.jumpstart.model.Model.deploy")
+    @mock.patch("sagemaker.jumpstart.factory.model.JUMPSTART_DEFAULT_REGION_NAME", region)
+    def test_proprietary_model_endpoint(
+        self,
+        mock_model_deploy: mock.Mock,
+        mock_model_init: mock.Mock,
+        mock_get_model_specs: mock.Mock,
+        mock_session: mock.Mock,
+        mock_validate_model_id_and_get_type: mock.Mock,
+        mock_sagemaker_timestamp: mock.Mock,
+    ):
+        mock_model_deploy.return_value = default_predictor
+
+        mock_sagemaker_timestamp.return_value = "7777"
+
+        mock_validate_model_id_and_get_type.return_value = JumpStartModelType.PROPRIETARY
+        model_id, _ = "ai21-summarization", "*"
+
+        mock_get_model_specs.side_effect = get_spec_from_base_spec
+
+        mock_session.return_value = sagemaker_session
+
+        model = JumpStartModel(
+            model_id=model_id,
```

Comment:
what happens if you supply a version?

Reply:
Version is supported and required now, since we shouldn't select wildcard version based on non-semantic version.